pkg/verification/service/verification_service.go

```diff
@@ -69,33 +67,29 @@ func NewVerificationService(context servicecontext.ServiceContext, cfg ServiceCo
 func (s *ServiceImpl) InitVerification(ctx *gin.Context, userID string, e164PhoneNumber string) error {
 	signup, err := s.Services().SignupService().GetUserSignup(userID)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
+		if errors.IsNotFound(err) {
 			log.Error(ctx, err, "usersignup not found")
-			return errors.NewNotFoundError(err, "usersignup not found")
+			return errors.NewNotFound(schema.GroupResource{Group: signup.APIVersion, Resource: signup.Kind}, userID)
 		}
 		log.Error(ctx, err, "error retrieving usersignup")
-		return errors.NewInternalError(err, fmt.Sprintf("error retrieving usersignup: %s", userID))
+		return err
 	}
 
 	// check that verification is required before proceeding
 	if signup.Spec.VerificationRequired == false {
 		log.Info(ctx, fmt.Sprintf("phone verification attempted for user without verification requirement: %s", userID))
-		return errors.NewBadRequest("forbidden request", "verification code will not be sent")
+		return errors.NewBadRequest("forbidden request:verification code will not be sent")
 	}
 
 	// Check if the provided phone number is already being used by another user
 	err = s.Services().SignupService().PhoneNumberAlreadyInUse(userID, e164PhoneNumber)
 	if err != nil {
-		switch t := err.(type) {
-		case *errors.Error:
-			if t.Code == http.StatusForbidden {
-				log.Errorf(ctx, err, "phone number already in use, cannot register using phone number: %s", e164PhoneNumber)
-				return errors.NewForbiddenError("phone number already in use", fmt.Sprintf("cannot register using phone number: %s", e164PhoneNumber))
-			}
-		default:
-			log.Error(ctx, err, "error while looking up users by phone number")
-			return errors.NewInternalError(err, "could not lookup users by phone number")
+		if errors.IsForbidden(err) {
+			log.Errorf(ctx, err, "phone number already in use, cannot register using phone number: %s", e164PhoneNumber)
+			return errors.NewForbidden(schema.GroupResource{}, "", fmt.Errorf("phone number already in use: cannot register using phone number: %s", e164PhoneNumber))
 		}
+		log.Error(ctx, err, "error while looking up users by phone number")
```

Comment:
I think we still should return 500 if the error is not forbidden.

Reply:
@alexeykazakov agreed, it will be returned as 500 here: https://github.com/codeready-toolchain/registration-service/blob/54c32bb024040def5b051da4d33998e8bf07c015/pkg/errors/error.go#L30